pkg/cephfs/volumeoptions.go

```diff
@@ -240,6 +257,10 @@ func newVolumeOptionsFromVolID(ctx context.Context, volID string, volOpt, secret
 			return nil, nil, err
 		}
 
+		if err = extractOptionalOption(&volOptions.SubvolumeGroup, "subvolumeGroup", volOpt); err != nil {
```

Comment:
`volOpt` is nil, when called from DeleteVolume, hence we will assume `SubvolumeGroup` as the default (as it is missing). As a result Delete would not work if the group name is stored as part of volume options, which in turn, partly comes from the StorageClass.